ebean-migration/src/main/java/io/ebean/migration/runner/LocalMigrationResources.java

```diff
@@ -44,42 +44,68 @@ boolean readInitResources() {
   }
 
   /**
-   * Read all the migration resources (SQL scripts) returning true if there are versions.
+   * Read all the migration resources (SQL scripts and JDBC migrations) returning true if there are versions.
    */
   boolean readResources() {
     if (readFromIndex()) {
       // automatically enable earlyChecksumMode when using index file with pre-computed checksums
       migrationConfig.setEarlyChecksumMode(true);
-      return true;
+    } else {
+      readResourcesForPath(migrationConfig.getMigrationPath());
     }
-    return readResourcesForPath(migrationConfig.getMigrationPath());
+    readJdbcMigrations();
+    Collections.sort(versions);
+    return !versions.isEmpty();
   }
 
+  /**
+   * Returns true, if an index file was found. Although, if file was empty, so we do not fall back
+   * to classpath scan!
+   */
   private boolean readFromIndex() {
     final var base = "/" + migrationConfig.getMigrationPath() + "/";
     final var basePlatform = migrationConfig.getBasePlatform();
     final var indexName = "idx_" + basePlatform + ".migrations";
     URL idx = resource(base + indexName);
     if (idx != null) {
-      return loadFromIndexFile(idx, base);
+      loadFromIndexFile(idx, base);
+      return true;
     }
     idx = resource(base + basePlatform + '/' + indexName);
     if (idx != null) {
-      return loadFromIndexFile(idx, base + basePlatform + '/');
+      loadFromIndexFile(idx, base + basePlatform + '/');
+      return true;
     }
     final var platform = migrationConfig.getPlatform();
     idx = resource(base + platform + indexName);
     if (idx != null) {
-      return loadFromIndexFile(idx, base + platform + '/');
+      loadFromIndexFile(idx, base + platform + '/');
+      return true;
     }
     return false;
   }
 
+
+  private void readJdbcMigrations() {
+    if (jdbcMigrationFactory != null) {
+      Iterator<JdbcMigration> iterator = jdbcMigrationFactory.iterator();
+      while (iterator.hasNext()) {
+        JdbcMigration jdbcMigration = iterator.next();
+        if (jdbcMigration.matches(migrationConfig)) {
+          final var version = MigrationVersion.parse(jdbcMigration.getName());
```

Comment:
Not sure that getName() should be used for version. I'm thinking there should be an explicit getVersion() method on the JdbcMigration interface (that implementations must provide a version).